scripts/skills/special/rf_first_turn_initiative.nut

```diff
@@ -0,0 +1,71 @@
+this.rf_first_turn_initiative <- ::inherit("scripts/skills/skill", {
+	m = {
+		// Public
+		InitiativePerAP = 40.0,	// For every this much Initiative an entity gains 1 AP. The received bonus can be negative
+		FlatAPBonus = -2,	// This is to balance out this new bonus for most entities
+
+		// Const
+		IconPositive = "ui/traits/trait_icon_46.png",	// Impatient Trait
+		IconNegative = "ui/traits/trait_icon_25.png",	// Hesitant Trait
+		TotalMinimum = -3,
+		TotalMaximum = 7,	// Maximum is just in case so that enemies that accidentally have extreme initiative don't one-shot you Turn one
+
+		// Private
+		CurrentAPBonus = 0,
+	},
+	function create()
+	{
+		this.m.ID = "special.rf_first_turn_initiative";
+		this.m.Name = "First Turn Initiative";
+		this.m.Description = "During the first turn of every combat, this character\'s [Action Points|Concept.ActionPoints] are modified depending on their [Initiative|Concept.Initiative] at that point.";
+		this.m.Icon = "ui/traits/trait_icon_46.png";
+		this.m.Type = ::Const.SkillType.StatusEffect;
+		this.m.Order = ::Const.SkillOrder.Last + 100;		// Some effects may set the Action Points to a flat value instead of increasing them, like 'possessed_undead_effect'
```

Comment:
I don't like the `+ 100` part. I'd just stick to `Last`. We can deal with potential edge cases when they happen.